### ISSUE TITLE
APSIMX function add_IncorpFOMPool method does not work

### ISSUE THREAD

**Issue by MAGALA-RICHARD** [APSIMInitiative/ApsimX#7446]
Hello, I am trying to write a manure management script using C# in apsim next generation, but the add_IncorpFOMPool method does not work. I want to incorporate the different pools of lignin cellulose and carbohydrate from the manure into the soil. 
When I write the Carbon nitrogen ratio, also it does not work. It says it is read-only. Please see attached file for any help please

[add manure to a specific crop on sowing in a rotation.txt](https://github.com/APSIMInitiative/ApsimX/files/9347756/add.manure.to.a.specific.crop.on.sowing.in.a.rotation.txt)

**Comment by hol353** [comment]
If you .zip your .apsimx file and attach to this issue I'll have a look. Your manager script looks ok to me. When you say it doesn't work do you mean it gives an error or something else?

**Comment by MAGALA-RICHARD** [comment]
Hello, it brings an error, and the script is not compiled when that method
is called.

I am guessing am used the method wrongly but followed the code pomp prompt
of how the method is used

**Comment by hol353** [comment]
The script compiles correctly for me. I need to see your .apsimx file. Can you zip it and attach to the GitHub issue?

Can you also tell us what the error message is? Click on 'More information' button beside the error message and copy and paste the full error message to this GitHub issue.

**Comment by MAGALA-RICHARD** [comment]
Here is the error that comes up. I have attached the APSIMX file as well.

Thank you!

System.Exception: Errors found in manager model AddManureRotation
APSIM917f4d10-7e30-4bfe-8f7f-8757d53cca59.cs(57,21): error CS0571: 'SurfaceOrganicMatter.IncorpFOMPool.add': cannot explicitly call operator or accessor

   at Models.Manager.RebuildScriptModel() in C:\Users\ContainerAdministrator\AppData\Local\Temp\ApsimX\Models\Management\Manager.cs:line 158
   at UserInterface.Presenters.ManagerPresenter.BuildScript() in C:\Users\ContainerAdministrator\AppData\Local\Temp\ApsimX\ApsimNG\Presenters\ManagerPresenter.cs:line 202
[add_manureTrial.zip](https://github.com/APSIMInitiative/ApsimX/files/9465485/add_manureTrial.zip)

**Comment by hol353** [comment]
To get your simulation to run I had to remove this line:
```
SOM.add_IncorpFOMPool(SOM);
```

from your AddManagerRotation and change your cultivar. OPV isn't a maize cultivar in the released APSIM. The simulation then ran ok. The summary file reports that manure is being added.

**Comment by MAGALA-RICHARD** [comment]
Sorry OPV was an edited cultvar in the replacement module, which I forgot to attach to the apsimx file. Yes, when you remove that line it runs, but am just wondering whether it is not neccessary or not to call that method.

**Comment by MAGALA-RICHARD** [comment]
When you remove that line, it runs good, but I don't think the SOM is being incorporated into he soil according to the soil organic carbon output

**Comment by hol353** [comment]
You are calling ```SOM.AddFaeces(Data);``` which should be all you need to do.

The best way to determine whether it works is to produce a daily output file (rather than yearly) and look at the SurfaceOrganicMatter.Wt as well as NO3 by layer i.e. do some detective work.

**Comment by MAGALA-RICHARD** [comment]
Thank you. This is a good diagnostic idea, but the results show that for manure application, the surface organic matter remains at the surface if we incorporate all the residues in contrast to the control, where I used synthetic fertilizers. Ideally, if the surface organic matter is incorporated, all residues are distributed evenly in the specified depth. They will be less left on the surface. This is the reason why I was calling that method. This was confirmed when I halved the fraction of residues incorporated into the soil. In this case, the control got the surface residues reflected as surface organic matter. The control had higher carbon than the manure application experiment.
